lib/elixir/lib/enum.ex

```diff
@@ -969,10 +969,18 @@ defmodule Enum do
     do_into(enumerable, collectable)
   end
 
+  def into(enumerable, %{__struct__: MapSet, map: map}) when map_size(map) == 0 do
```

Comment:
The only way to make the changes to `Map.new` and `MapSet.new` benefit `Enum.into` is when `Enum.into` is being used to create a new map / mapset. If it's being used to update an existing one you would have to first do `:maps.to_list` on the existing map, prepend items, and then convert back to a map. Unless the number of items you're adding is substantially larger than the number of items already in the map the overhead won't be worth it.

This is why it seemed like the only sure case we can optimize is when using Enum.into to build a new map or mapset.